numpy/_typing/_add_docstring.py

```diff
@@ -149,4 +149,45 @@ def _parse_docstrings() -> str:
 
     """)
 
+add_newdoc('Array', repr(Array),
+    """
+    A `np.ndarray[tuple[*Ts], np.dtype[+ScalarType]] <numpy.ndarray>` type
+    alias :term:`generic <generic type>` w.r.t. its shape and
+    `dtype.type <numpy.dtype.type>`.
+
+    Can be used during runtime for typing arrays with a given dtype
+    and specified shape.  Particularly useful for functions which modify number
+    of axes.  If it is important to enforce integer axis sizes, use
+    `np.ndarray <numpy.ndarray>` for typing.  If axes dimensions are
```

Comment:
```suggestion
    `np.ndarray <numpy.ndarray>` for typing.  If axes dimensions are
```
I'd maybe drop the reference to `np.ndarray` and the mention of enforcing integer axis sizes. This feels a bit too much like discussing internal implementation details for it to belong in the general docs (well, those associated with upstream limitations anyway...).